File: examples/v2/stack-tags/go/main.go
package main

import (
	stacks "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/stacks"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "v2-stack-tags-example"
		}
		stackName := cfg.Get("stackName")
		if stackName == "" {
			stackName = "dev"
		}
		tagValue := cfg.Get("tagValue")
		if tagValue == "" {
			tagValue = "v2-tag-value"
		}

		parentStack, err := stacks.NewStack(ctx, "parentStack", &stacks.StackArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
		})
		if err != nil {
			return err
		}

		if _, err := stacks.NewTag(ctx, "ownerTag", &stacks.TagArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: parentStack.ProjectName,
			StackName:   parentStack.StackName,
			Name:        pulumi.String("owner"),
			Value:       pulumi.String("pulumicloud-v2-example"),
		}); err != nil {
			return err
		}

		if _, err := stacks.NewTag(ctx, "customTag", &stacks.TagArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: parentStack.ProjectName,
			StackName:   parentStack.StackName,
			Name:        pulumi.String("purpose"),
			Value:       pulumi.String(tagValue),
		}); err != nil {
			return err
		}

		ctx.Export("parent", parentStack.ID())
		return nil
	})
}

File: examples/v2/stack-tags/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
project_name = config.get("projectName") or "v2-stack-tags-example"
stack_name = config.get("stackName") or "dev"
tag_value = config.get("tagValue") or "v2-tag-value"

parent_stack = ps_v2.stacks.Stack(
    "parentStack",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
)

ps_v2.stacks.Tag(
    "ownerTag",
    org_name=service_org,
    project_name=parent_stack.project_name,
    stack_name=parent_stack.stack_name,
    name="owner",
    value="pulumicloud-v2-example",
)

ps_v2.stacks.Tag(
    "customTag",
    org_name=service_org,
    project_name=parent_stack.project_name,
    stack_name=parent_stack.stack_name,
    name="purpose",
    value=tag_value,
)

pulumi.export("parent", parent_stack.id)
